From 2c024cf701ee54cc386cfa5bf6fda566214972c4 Mon Sep 17 00:00:00 2001
From: Pawel Chojnacki <pawel@chojnacki.ws>
Date: Tue, 16 Feb 2016 16:49:45 +0100
Subject: [PATCH] Wikia Consul tunings

---
 consul/config.go | 8 ++++++++
 version.go       | 2 +-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/consul/config.go b/consul/config.go
index 78b3fc8..3dcdefc 100644
--- a/consul/config.go
+++ b/consul/config.go
@@ -292,6 +292,14 @@ func DefaultConfig() *Config {
 	conf.SerfLANConfig.ReconnectTimeout = 3 * 24 * time.Hour
 	conf.SerfWANConfig.ReconnectTimeout = 3 * 24 * time.Hour
 
+	// https://www.serfdom.io/docs/internals/simulator.html
+	conf.SerfLANConfig.MemberlistConfig.ProbeInterval = 30 * time.Second
+	conf.SerfLANConfig.MemberlistConfig.ProbeTimeout = 2 * time.Second // increase the probe timeout to 2s
+	conf.SerfLANConfig.MemberlistConfig.GossipNodes = 4                // Gossip to 4 nodes instead of 3. At the expense of higher bandwidth (2x) decrease convergence time
+	conf.SerfLANConfig.MemberlistConfig.RetransmitMult = 5             // Retransmit messages 5*log(N+1)
+	conf.SerfLANConfig.MemberlistConfig.SuspicionMult = 20             // Suspect a node for 20 * log(N+1) * GossipInterval
+	conf.SerfLANConfig.RecentIntentBuffer = 512                        // Increase from 128 to allow for better filtering of known messages
+
 	// WAN Serf should use the WAN timing, since we are using it
 	// to communicate between DC's
 	conf.SerfWANConfig.MemberlistConfig = memberlist.DefaultWANConfig()
diff --git a/version.go b/version.go
index b6b9b2b..ac0179c 100644
--- a/version.go
+++ b/version.go
@@ -7,7 +7,7 @@ var (
 )
 
 // The main version number that is being run at the moment.
-const Version = "0.6.3"
+const Version = "0.6.3-wikia1"
 
 // A pre-release marker for the version. If this is "" (empty string)
 // then it means that it is a final release. Otherwise, this is a pre-release
-- 
2.5.0

